lambdas/api-handler/src/handlers/__tests__/get-letter-data.test.ts

```diff
@@ -0,0 +1,115 @@
+// mock error mapper
+jest.mock('../../mappers/error-mapper');
+import { mapErrorToResponse } from '../../mappers/error-mapper';
+const mockedMapErrorToResponse = jest.mocked(mapErrorToResponse);
+const expectedErrorResponse: APIGatewayProxyResult = {
+  statusCode: 400,
+  body: 'Error'
+};
+mockedMapErrorToResponse.mockReturnValue(expectedErrorResponse);
+
+// mock letterService
+jest.mock('../../services/letter-operations');
+import * as letterService from '../../services/letter-operations';
+
+import type { APIGatewayProxyResult, Context } from 'aws-lambda';
+import { mockDeep } from 'jest-mock-extended';
+import { makeApiGwEvent } from './utils/test-utils';
+import { ValidationError } from '../../errors';
+import * as errors from '../../contracts/errors';
+import { createGetLetterDataHandler } from '../get-letter-data';
+import { S3Client } from '@aws-sdk/client-s3';
+import pino from 'pino';
+import { LetterRepository } from '../../../../../internal/datastore/src';
+import { EnvVars } from '../../config/env';
+import { Deps } from "../../config/deps";
+
+describe('API Lambda handler', () => {
+
+  const mockedDeps: jest.Mocked<Deps> = {
+    s3Client: {} as unknown as S3Client,
+    letterRepo: {} as unknown as LetterRepository,
+    logger: { info: jest.fn(), error: jest.fn() } as unknown as pino.Logger,
+    env: {
+      SUPPLIER_ID_HEADER: 'nhsd-supplier-id',
+      APIM_CORRELATION_HEADER: 'nhsd-correlation-id',
+      LETTERS_TABLE_NAME: 'LETTERS_TABLE_NAME',
+      LETTER_TTL_HOURS: 1,
+      DOWNLOAD_URL_TTL_SECONDS: 1
+    } as unknown as EnvVars
+  }
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('returns 303 Found with a pre signed url', async () => {
```

Comment:
302 is Found, 303 is See Other

I'm not sure that it matters much for the functionality of the API, but I would probably go with 302 Found as that matches semantically with what we're doing.

[MDN lists 303](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status/303) as more often used to redirect from a POST or PUT to the resulting updated resource, although in our current implementation we just return the resulting record directly rather than redirecting.

Reply:
Good catch. We had discussed this with Marks S and decided to go with a 303, but happy to change it to 302 if we want to? That might need a ticket though